flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -1410,6 +1410,26 @@ object AggregateUtil {
         case _: SqlCountAggFunction =>
           aggregates(index) = new CountAggFunction
 
+        case collect: SqlAggFunction if collect.getKind == SqlKind.COLLECT =>
+          aggregates(index) = sqlTypeName match {
+            case TINYINT =>
+              new ByteCollectAggFunction
+            case SMALLINT =>
+              new ShortCollectAggFunction
+            case INTEGER =>
+              new IntCollectAggFunction
+            case BIGINT =>
+              new LongCollectAggFunction
+            case VARCHAR | CHAR =>
+              new StringCollectAggFunction
+            case FLOAT =>
+              new FloatCollectAggFunction
+            case DOUBLE =>
+              new DoubleCollectAggFunction
+            case _ =>
+              new ObjectCollectAggFunction
+          }
+
```

Comment:
we need to set `accTypes(index) = aggregates(index).getAccumulatorType` in order to activate the `MapView` feature.